### PR TITLE
JP-2909: Expanding Sequence Number Field

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**
Resolves [JP-2909](https://jira.stsci.edu/browse/JP-2909)

Closes #

This PR expands the ASN file names sequence number fields from 3 characters to 5 characters, allowing for association files in a sequence.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -19,6 +19,9 @@ associations
   exception if path data is found.  Also, expanded documentation to make this
   more clear to users. [#7008]
 
+- Expand the sequence field in a file name for association files from
```
Comment (hbushouse) [inline CHANGES.rst:22]: This needs to get moved up to the 1.8.1 section

**Review by hbushouse — COMMENTED** [review]
Changes look OK to me, other than fixing the change log. Are we hoping to include this in the final B9.0 build?

**Comment by hbushouse** [comment]
I think I remember you mentioning results of regression tests, but I don't see any test runs in "JWST-Developers-Pull-Requests". Did you just run them locally or what?

**Comment by kmacdonald-stsci** [comment]
> I think I remember you mentioning results of regression tests, but I don't see any test runs in "JWST-Developers-Pull-Requests". Did you just run them locally or what?

I always do regression tests locally to analyze any failures and errors.  I think the failures and errors I'm seeing are due to other things.  I'm going to rebase, then run a Jenkins regression test.

**Review by stscieisenhamer — APPROVED** [review]
Concerning getting into 9.0, there are a couple considerations:

1. Files have already been created with 4 digits, and nothing has apparently broken. So, this is not a critical item.
2. Though nothing appears "broken" in ground processing, downstream services, such as archive, should be alerted to ensure there are no issues.

I suspect it will not be an issue, since files with at least 4 digits have been created and archived. This fix is to fix the inconsistency of the padding.

**Comment by hbushouse** [comment]
Good point about alerting the archive folks. We have a B9 tag-up this afternoon, where this can be mentioned.

**Comment by kmacdonald-stsci** [comment]
The Jenkins test has expected failures due to file name mismatches since this PR specifically changes output association file names.

**Comment by hbushouse** [comment]
The 3 differences detected in the regtest run at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/441/ look to be expected (more leading zeros in an ASN version number).

**Review by hbushouse — APPROVED** [review]
Looks good, but let's hold off on merging this until we get confirmation from CAOM folks that the change in file name syntax won't cause any problems for them.

**Review by hbushouse — APPROVED** [review]

**Pull request merged**